p2p/libp2p/module.go

```diff
@@ -0,0 +1,338 @@
+	"github.com/pokt-network/pocket/p2p/libp2p/identity"
+	"github.com/pokt-network/pocket/p2p/libp2p/protocol"
+	"github.com/pokt-network/pocket/p2p/providers"
+	"github.com/pokt-network/pocket/p2p/providers/addrbook_provider"
+	"github.com/pokt-network/pocket/p2p/providers/addrbook_provider/persistence"
+	"github.com/pokt-network/pocket/p2p/stdnetwork"
+	typesP2P "github.com/pokt-network/pocket/p2p/types"
+	"github.com/pokt-network/pocket/runtime/configs"
+	poktCrypto "github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/messaging"
+	"github.com/pokt-network/pocket/shared/modules"
+)
+
+var _ modules.P2PModule = &libp2pModule{}
+
+type libp2pModule struct {
+	log                   *modules.Logger
+	bus                   modules.Bus
+	cfg                   *configs.P2PConfig
+	addrBookProvider      providers.AddrBookProvider
+	currentHeightProvider providers.CurrentHeightProvider
+	identity              libp2p.Option
+	listenAddrs           libp2p.Option
+	host                  host.Host
+	pubsub                *pubsub.PubSub
+	topic                 *pubsub.Topic
+	subscription          *pubsub.Subscription
+	network               typesP2P.Network
+}
+
+// NewReadStreamDeadline returns a future deadline
+// based on the read stream timeout duration.
+func NewReadStreamDeadline() time.Time {
+	return time.Now().Add(readStreamTimeoutDuration)
+}
+
+var (
+	// TODO: consider security exposure of and "safe minimum" for timeout.
+	// TODO: parameterize and expose via config.
+	// ReadStreamTimeout is the duration to wait for a read operation on a
+	// stream to complete, after which the stream is closed ("timed out").
+	readStreamTimeoutDuration = time.Second * 10
+	// ErrModule wraps errors which occur within the libp2pModule implementation.
+	ErrModule = common.NewErrFactory("LibP2P module error")
+	// ErrCloseStream wraps errors which occur when attempting to close a peer stream.
+	ErrCloseStream = common.NewErrFactory("an error occurred while closing peer stream")
+)
+
+func Create(bus modules.Bus) (modules.Module, error) {
+	return CreateWithProviders(
+		bus,
+		persistence.NewPersistenceAddrBookProvider(bus),
+		bus.GetConsensusModule(),
+	)
+}
+
+func CreateWithProviders(bus modules.Bus, addrBookProvider addrbook_provider.AddrBookProvider, currentHeightProvider providers.CurrentHeightProvider) (modules.Module, error) {
+	return new(libp2pModule).CreateWithProviders(bus, addrBookProvider, currentHeightProvider)
+}
+
+func (mod *libp2pModule) GetModuleName() string {
+	// TODO: double check if this should change.
+	return modules.P2PModuleName
+}
+
+func (mod *libp2pModule) Create(bus modules.Bus) (modules.Module, error) {
+	return Create(bus)
+}
+
+func (mod *libp2pModule) CreateWithProviders(bus modules.Bus, addrBookProvider addrbook_provider.AddrBookProvider, currentHeightProvider providers.CurrentHeightProvider) (modules.Module, error) {
+	*mod = libp2pModule{
+		log:                   new(modules.Logger),
```

Comment:
Please see the pattern we follow for loggers: https://github.com/pokt-network/pocket/wiki/Logger-Readme#logger-initialization

1. Name
2. Initialization with namespace
3. If we go with a pointer, we should add a TODO or update the documentation for that to be the case everywhere

Reply:
I did read that one but see now that I missed a couple of details. Thanks for surfacing this!

The reason for the change to a pointer is (as @deblasis [pointed out](https://github.com/pokt-network/pocket/pull/502#issuecomment-1424590635)) is that the linter identifies it as a `hugeParam` otherwise.

More than happy to add a TODO or just update the docs if you prefer.

Reply:
Please add a TODO in the logger module and/or the logger README